src/hfold/Parser.hs

```diff
@@ -0,0 +1,55 @@
+module Hfold.Parser where
+
+import Data.Monoid ((<>))
+import Data.Text (Text)
+import qualified Data.Text as T
+import Text.ParserCombinators.Parsec
+
+import Hfold.Types
+
+parseFile :: Text -> [Content]
+parseFile t = case parse parseContent "content" (T.unpack t) of
+    Left err -> []
+    Right xs -> xs
+
+parseContent :: Parser [Content]
+parseContent = do
+    c <- many (quoted <|> codeBlock <|> normal)
```

Comment:
I constantly tried to use applicative style in this and couldn't ever get it to work. This one works though. Nice.